src/resonate/dst/scheduler.py

```diff
@@ -310,7 +326,36 @@ def _route_fn_or_coroutine(
 
     def _create_promise(self, promise_id: str, action: Invoke | Sleep) -> Promise[Any]:
```

Comment:
Extending the `_create_promise` function. Now it also creates the durable promise record. If, when created, it is already completed then I immediately resolve the promise to enable shortcutting in the future